packages/key-management/src/util/ownSignatureKeyPaths.ts

```diff
@@ -25,12 +71,11 @@ export const ownSignatureKeyPaths = async (
     ).filter(isNotNil)
   ).map(({ type, index }) => ({ index, role: Number(type) }));
 
-  const rewardAccounts = new Set(knownAddresses.map(({ rewardAccount }) => rewardAccount));
+  const rewardAccounts = [...new Set(knownAddresses.map(({ rewardAccount }) => rewardAccount))];
 
-  const isStakingKeySignatureRequired = txBody.certificates?.length;
   if (
-    isStakingKeySignatureRequired ||
-    txBody.withdrawals?.some((withdrawal) => rewardAccounts.has(withdrawal.stakeAddress))
+    isStakingKeySignatureRequired(rewardAccounts, txBody.certificates) ||
+    txBody.withdrawals?.some((withdrawal) => rewardAccounts.includes(withdrawal.stakeAddress))
```

Comment:
`isStakingKeySignatureRequired` could also check `withdrawals`, as it is also signing with stake key